tests/test_integration_http.py

```diff
@@ -611,6 +611,114 @@ def test_capture_minidump(cmake, httpserver):
     assert_minidump(envelope)
 
 
+@pytest.mark.parametrize("port_correct", [True, False])
+def test_proxy_from_env(cmake, httpserver, port_correct):
+    # TODO how can we test this? If it doesn't get read but it's there, we don't know (since it'll just fallback)
+    #     can we listen to mitmdump's port and see if it gets traffic?
```

Comment:
We now have a way to read the stdout after terminating `mitmdump`, and we can check it as such:
```c
proxy_process.terminate()
proxy_process.wait()
stdout, stderr = proxy_process.communicate()
assert "POST" in stdout
```
It makes sense to have this check for all of our proxy tests, as this is a more rigorous check on what goes through/around our proxy server.